src/libostree/ostree-repo.c

```diff
@@ -3292,6 +3292,28 @@ ostree_repo_get_mode (OstreeRepo  *self)
   return self->mode;
 }
 
+/**
+ * ostree_repo_get_min_free_space:
+ * @self: Repo
+ * @error: error
+ *
+ * Before this function can be used, ostree_repo_prepare_transaction() must have been
+ * called.
+ *
+ * Returns: Value(in bytes) of min-free-space-* config option
+ */
+guint64
+ostree_repo_get_min_free_space (OstreeRepo  *self, GError **error)
+{
+  g_return_val_if_fail (OSTREE_IS_REPO (self), 0);
+
+  g_assert (self->txn.blocksize != 0);
+  if (self->reserved_blocks > G_MAXUINT64 / self->txn.blocksize)
+    return glnx_throw (error, "min-free-space(in bytes) value was too high");
```

Comment:
It seems a bit odd that the code in `ostree_repo_prepare_transaction()` doesn’t check this as well. (I realise it’s because that code operates in terms of blocks rather than bytes.) It would be weird for a transaction to happily go ahead, but then return an error only when you call `ostree_repo_get_min_free_space()`.

Reply:
hmm, this makes sense but I think let's file a separate issue and PR for that?

Reply:
I suggest it’s handled here, because there is no inconsistency before this PR is merged, and there is an inconsistency afterwards.